unit/java_bytecode/java_types/generic_type_index.cpp

```diff
@@ -0,0 +1,92 @@
+/*******************************************************************\
+
+ Module: Unit tests for java_types
+
+ Author: Diffblue Ltd.
+
+\*******************************************************************/
+
+#include <testing-utils/catch.hpp>
+#include <java_types.h>
+
+SCENARIO("generic_type_index", "[core][java_types]")
+{
+  GIVEN("Generic type LGenericClass<TX;TY;>; and parameters X, Y, Z")
+  {
+    const auto symbol_type = symbol_typet("MyType");
+    const auto generic_symbol_type = java_generic_symbol_typet(
+      symbol_type, "LGenericClass<TX;TY;>;", "PrefixClassName");
+    java_generic_parametert paramX("PrefixClassName::X", symbol_typet());
+    java_generic_parametert paramY("PrefixClassName::Y", symbol_typet());
+    java_generic_parametert paramZ("PrefixClassName::Z", symbol_typet());
+
+    WHEN("Looking for parameter indexes")
+    {
+      const auto indexX = generic_symbol_type.generic_type_index(paramX);
+      const auto indexY = generic_symbol_type.generic_type_index(paramY);
+      const auto indexZ = generic_symbol_type.generic_type_index(paramZ);
+
+      THEN("X has index 0, Y index 1 and Z is not found")
+      {
+        REQUIRE(indexX.has_value());
+        REQUIRE(indexX.value() == 0);
+        REQUIRE(indexY.has_value());
+        REQUIRE(indexY.value() == 1);
+        REQUIRE_FALSE(indexZ.has_value());
+      }
+    }
+  }
+
+  GIVEN("Generic type LGenericClass<Tkey;Tvalue;>; and"
+          " parameters key, value, x")
```

Comment:
How is this example different from the previous one?

Reply:
It has non-single letter generic parameter names as suggested by @thk123

Reply:
I see. Maybe you can combine them into one example though, with one single letter and one non-single letter case.